### PR TITLE
bug/5520 ComboBox with numeric key zero does not work

### LINKED ISSUE #5520: ComboBox with numeric key zero does not work
### Bug Report 
 
- __Package version(s)__: 6.23.0
 
#### Priorities and help requested:
 
Are you willing to submit a PR to fix? (No)
 
Requested priority: (High) 
 
#### Describe the issue:
 
The ComboBox is not usable with items with an numeric key of zero.
 
#### Actual behavior:
 
ComboBox is blank when `selectedKey={0}`.
 
#### Expected behavior:
 
A ComboBox with an entry with numeric key 0 should display the item.
 
### If applicable, please provide a codepen repro:

https://codepen.io/anon/pen/gjbdgM

### PR DESCRIPTION
**Pull request by a-tarasyuk**
- [x] Fixes #5520
- [x] Include a change request file using `$ npm run change`

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx [hunk 1]
```diff
@@ -1924,31 +1927,32 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
...
+
+    return this._buildSelectedKeys(selectedKey);
+  }
+
+  private _buildSelectedKeys(selectedKey: string | number | string[] | number[] | undefined): string[] | number[] {
+    if (typeof selectedKey === 'string' && selectedKey !== '') {
+      return [selectedKey];
+    }
+
+    if (typeof selectedKey === 'number' && selectedKey >= 0) {
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: this is the weirdest bit of code I've seen. I don't have original context on this. @jspurlin do you ?

What is the intention here? It seems like it's don't nothing except filtering selectedKey values it probably shouldn't be!

Shouldn't this just be:

```
if (selectedKey === undefined) {
return [];
}

return (selectedKey instanceof Array ? selectedKey : [ selectedKey ]);
}
```

Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: (Also, I don't really understand why we'd special case blank strings. If the user passes in a blank string, they did something weird.)

**Comment by a-tarasyuk** [comment]
@dzearing thanks, closed!

**Comment by a-tarasyuk** [comment]
I don't know why, however I don't see the following comment anymore,

> this is the weirdest bit of code I've seen. I don't have original context on this. @jspurlin do you ?

Re-published. Maybe it will be usefuls to others, :)

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx (hunk 1 above)
Comment (jspurlin) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: I think this code was added when multiselect comboBox functionality was added

---
In reply to: [205835697](https://github.com/OfficeDev/office-ui-fabric-react/pull/5705#discussion_r205835697) [](ancestors = 205835697)

Comment (a-tarasyuk) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: @jspurlin thank you very much for the clarification 👍 :)

**Comment by dzearing** [comment]
@a-tarasyuk hope i did t give the wrong impression. I was confused on the purpose of the function, so i was asking @jspurlin for context. I reworded the text to be less pointed. (After rereading it i thought it sounded too angry ;)

I still thin the fix to support 0 as a key is a great fix!

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx (hunk 1 above)
Comment (a-tarasyuk) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: @dzearing I think, your suggestion will cause the type error, [Example, need to enable `strictNullChecks`](https://www.typescriptlang.org/play/#src=class%20ComboBox%20%7B%0D%0A%20%20%20%20private%20_buildSelectedKeys(selectedKey%3A%20string%20%7C%20number%20%7C%20string%5B%5D%20%7C%20number%5B%5D%20%7C%20undefined)%3A%20string%5B%5D%20%7C%20number%5B%5D%20%7B%0D%0A%20%20%20%20%20%20%20%20if%20(typeof%20selectedKey%20%3D%3D%3D%20'string')%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20return%20%5B%5D%3B%0D%0A%20%20%20%20%20%20%20%20%7D%0D%0A%0D%0A%20%20%20%20%20%20%20%20return%20(selectedKey%20instanceof%20Array%20%3F%20selectedKey%20%3A%20%5BselectedKey%5D)%3B%0D%0A%20%20%20%20%7D%0D%0A%7D)

Maybe that's why the previous version had code dublication

```
let retKeys: string[] | number[] = [];

if (defaultSelectedKey) {
  if (defaultSelectedKey instanceof Array) {
    retKeys = defaultSelectedKey;
  } else if (typeof defaultSelectedKey === 'string') {
    retKeys = [defaultSelectedKey as string];
  } else if (typeof defaultSelectedKey === 'number') {
    retKeys = [defaultSelectedKey as number];
  }
} else if (selectedKey) {
  if (selectedKey instanceof Array) {
    retKeys = selectedKey;
  } else if (typeof selectedKey === 'string') {
    retKeys = [selectedKey as string];
  } else if (typeof selectedKey === 'number') {
    retKeys = [selectedKey as number];
  }
}

return retKeys;
```

**Pull request closed without merging**